src/main/java/com/github/manolo8/darkbot/gui/tree/editors/ImageOpener.java

```diff
@@ -0,0 +1,72 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.config.ImageWrapper;
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import eu.darkbot.api.config.util.OptionEditor;
+import com.github.manolo8.darkbot.gui.utils.Strings;
+import eu.darkbot.api.config.ConfigSetting;
+
+import javax.swing.*;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import java.awt.*;
+import java.io.File;
+
+public class ImageOpener extends JButton implements OptionEditor<ImageWrapper> {
+    private JFileChooser fc;
+    private ImageWrapper image;
+
+    public ImageOpener(){
+        putClientProperty("JButton.buttonType", "square");
+        addActionListener(e -> {
+            if (fc == null) {
+                fc = new JFileChooser(new File(".")) {
+                    @Override
+                    protected JDialog createDialog(Component parent) throws HeadlessException {
+                        JDialog dialog = super.createDialog(parent);
+                        dialog.setAlwaysOnTop(true);
+                        return dialog;
+                    }
+                };
+                FileNameExtensionFilter filter = new FileNameExtensionFilter(
+                        "PNG, JPG & GIF Images", "png","jpeg","jpg", "gif");
+                fc.setFileFilter(filter);
+                fc.setAcceptAllFileFilterUsed(false);
+            }
+            if(image == null){
+                image = new ImageWrapper();
+            }
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+                image.setPath(fc.getSelectedFile().getAbsolutePath());
+                setText(Strings.fileName(image.getPath()));
+            }
+        });
+
+    }
+    @Override
+    public void setText(String text) {
+        super.setText(text);
+        setPreferredSize(new Dimension(getFontMetrics(getFont()).stringWidth(getText()) + 16, 0));
+    }
+
+    @Override
+    public JComponent getEditorComponent(ConfigSetting<ImageWrapper> configSetting) {
+        setEditing(configSetting.getValue());
+        return this;
+    }
+
+    private void setEditing(ImageWrapper image){
+        this.image = image; //do i need this, i dont think its needed
+        setText(image.getPath());
```

Comment:
I would simplify the text-representation of this, showing a whole system path is often impractical (too long)
You can probably just trim for the last N sections of the path based on a max aim length, say, 80 chars max, so instead of:
c:\Users\User\Desktop\Images\darkbot\image.jpg
it would maybe trim to:
...\Images\darkbot\image.jpg